### PR TITLE
Added simpleclient_log4j2 appender to record logging metrics

### PR DESCRIPTION
**Pull request by yngveh**
This PR add support for log4j2 appender metrics logging, This has the same functunality as the existing simpleclient_log4j

### CODE REVIEW

**Comment by brian-brazil** [comment]
Why do we need this? There doesn't appear to be log4j2.

**Comment by yngveh** [comment]
We need this when using log4j2 as logging, example using the spring-boot-starter-log4j2 maven artifact.

From pom.xml, log4j2 dependency is defined with
 ```    dependency>
            <groupId>org.apache.logging.log4j</groupId>
            <artifactId>log4j-core</artifactId>
            <version>2.7</version>
        </dependency>```

File: simpleclient_log4j2/src/main/java/io/prometheus/client/log4j2/InstrumentedAppender.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ *   log4j_appender_total{level="info",} 3.0
+ *   log4j_appender_total{level="warn",} 4.0
+ *   log4j_appender_total{level="error",} 5.0
+ *   log4j_appender_total{level="fatal",} 6.0
+ * </pre>
+ */
+@Plugin(name = "Prometheus", category = "Core", elementType = "appender")
+public final class InstrumentedAppender extends AbstractAppender {
+
+    public static final String COUNTER_NAME = "log4j_appender_total";
```
Comment (brian-brazil) [inline simpleclient_log4j2/src/main/java/io/prometheus/client/log4j2/InstrumentedAppender.java]: This name is already taken by the log4j module

File: simpleclient_log4j2/pom.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>io.prometheus</groupId>
+            <artifactId>simpleclient</artifactId>
+            <version>0.0.21-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
+            <version>2.7</version>
```
Comment (brian-brazil) [inline simpleclient_log4j2/pom.xml]: Use the oldest version you can, to ensure it remains backwards compatible

**Comment by brian-brazil** [comment]
This should be mentioned in the readme.

**Comment by yngveh** [comment]
Pushed suggested changes.

File: README.md [hunk 3]
```diff
@@ -243,6 +243,21 @@ To register the Logback collector can be added to the root level like so:
 
 </configuration>
 ```
+To register the log4j2 collector at root level:
```
Comment (brian-brazil) [inline README.md]: We should add log4j for completeness

**Comment by brian-brazil** [comment]
Can you look at the failing tests?

**Comment by yngveh** [comment]
The failing tests doesn't seem to be directly related to the last push.
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.8:aggregate (default-cli) on project parent: Execution default-cli of goal org.apache.maven.plugins:maven-javadoc-plugin:2.8:aggregate failed: Plugin org.apache.maven.plugins:maven-javadoc-plugin:2.8 or one of its dependencies could not be resolved: Could not transfer artifact org.apache.maven:maven-archiver:jar:2.3 from/to central (https://repo.maven.apache.org/maven2): handshake alert:  unrecognized_name -> [Help 1]
```

**Comment by brian-brazil** [comment]
Retried it, is passing now.

**Pull request merged**